### ISSUE TITLE
Released versions contain test data

### ISSUE THREAD

**Issue by weiznich** [BurntSushi/bstr#200]
While reviewing some updates to our dependencies I noticed that the last version (1.11.0) of `bstr` uploaded to crates.io contains unnecessary test data in `src/unicode/data`. While this is not a problem on it's own it is at least wasteful as these data need not to be there for having a correctly working crate.

Cargo allows to exclude such files via an entry in your Cargo.toml file: https://doc.rust-lang.org/cargo/reference/manifest.html#the-exclude-and-include-fields

You can see the included data here: https://diff.rs/bstr/1.10.0/1.11.0/

If you are interested in addressing that I'm happy to contribute a PR for this.

**Comment by BurntSushi** [comment]
It's always been there: https://diff.rs/bstr/1.9.0/1.10.0/src%2Funicode%2Fdata%2FGraphemeBreakTest.txt

I'm aware that such files can be excluded from the uploaded artifact, but tests use them: https://github.com/BurntSushi/bstr/blob/41f8bdb9ec08912a38a1b2da704fcf0a19fcea45/src/unicode/word.rs#L414-L428

If the data files are excluded, then you wouldn't be able to run tests using just the uploaded artifact. To be honest, I don't know how important that is, and it isn't something that I do personally. But I think some tooling (like `crater`?) does it. And it seems useful to be able to run tests on the thing you download from crates.io. With that said, I'm not religious about it, and if the data files were very large, I'd be open to excluded them. But in this case, the data files represent a marginal increase in total file size (because the collection of `regex-automata` DFA files is much bigger).

**Comment by weiznich** [comment]
I agree with you that this is not a big issue, given that these are readable text files.

I'm not aware of any easy way to execute tests for a crate downloaded from crates.io. You could manually copy the source code from `~/.cargo/` and afterwards run the tests there, but I'm not sure how common that usecase is. As far as I know  that's nothing people commonly do. Also there are quite a few crates that explicitly exclude their tests from the published code just to keep the code size smaller.

As for crater: I think crater also grabs the source code from github as well, so it would run tests there and not for published version.

That written: I'm also totally fine to just close this issue, I mostly filled it to see if you were interested in removing these files.

**Comment by BurntSushi** [comment]
`bstr 1.11.1` on crates.io now omits these files.

This is probably the right default. Let's see if someone complains. Then we'll have a real use case.

**Comment by weiznich** [comment]
Thanks for the fast release :heart:
